internal/server/change_case_status.go

```diff
@@ -41,33 +53,72 @@ func ChangeCaseStatus(client ChangeCaseStatusClient, tmpl template.Template) Han
 		}
 
 		data := changeCaseStatusData{
-			XSRFToken: ctx.XSRFToken,
-			Entity:    fmt.Sprintf("%s %s", cs.DigitalLpa.SiriusData.Subtype, caseUID),
-			CaseUID:   caseUID,
-			OldStatus: status,
-			NewStatus: postFormString(r, "status"),
+			XSRFToken:          ctx.XSRFToken,
+			Error:              sirius.ValidationError{Field: sirius.FieldErrors{}},
+			Entity:             fmt.Sprintf("%s %s", cs.DigitalLpa.SiriusData.Subtype, caseUID),
+			CaseUID:            caseUID,
+			OldStatus:          status,
+			NewStatus:          postFormString(r, "status"),
+			StatusChangeReason: postFormString(r, "statusReason"),
 		}
 
-		if r.Method == http.MethodPost {
-			caseStatusData := sirius.CaseStatusData{
-				Status: data.NewStatus,
+		data.StatusItems = []statusItem{
+			{Value: "draft", Label: "Draft", ConditionalItem: false},
+			{Value: "in-progress", Label: "In progress", ConditionalItem: false},
+			{Value: "statutory-waiting-period", Label: "Statutory waiting period", ConditionalItem: false},
+			{Value: "registered", Label: "Registered", ConditionalItem: false},
+			{Value: "suspended", Label: "Suspended", ConditionalItem: false},
+			{Value: "do-not-register", Label: "Do not register", ConditionalItem: false},
+			{Value: "expired", Label: "Expired", ConditionalItem: false},
+			{Value: "cannot-register", Label: "Cannot register", ConditionalItem: true},
+			{Value: "cancelled", Label: "Cancelled", ConditionalItem: true},
+			{Value: "de-registered", Label: "De-registered", ConditionalItem: false},
+		}
+
+		group.Go(func() error {
+			data.CaseStatusChangeReasons, err = client.RefDataByCategory(ctx.With(groupCtx), sirius.CaseStatusChangeReason)
+			if err != nil {
+				return err
 			}
 
-			err = client.EditDigitalLPAStatus(ctx, caseUID, caseStatusData)
+			return nil
+		})
 
-			if ve, ok := err.(sirius.ValidationError); ok {
-				w.WriteHeader(http.StatusBadRequest)
-				data.Error = ve
-			} else if err != nil {
-				return err
-			} else {
-				data.Success = true
+		if err := group.Wait(); err != nil {
+			return err
+		}
+
+		if r.Method == http.MethodPost {
+			if (data.NewStatus == "cannot-register" || data.NewStatus == "cancelled") && data.StatusChangeReason == "" {
 				data.OldStatus = data.NewStatus
+				w.WriteHeader(http.StatusBadRequest)
+				data.Error.Field["changeReason"] = map[string]string{
```

Comment:
This should be able to be handled in the backend. We have done similar logic in the create draft file. For the different addresses.